File: src/main/java/com/refinedmods/refinedstorage/screen/InterfaceScreen.java
package com.refinedmods.refinedstorage.screen;

import com.refinedmods.refinedstorage.RS;
import com.refinedmods.refinedstorage.blockentity.InterfaceBlockEntity;
import com.refinedmods.refinedstorage.blockentity.NetworkNodeBlockEntity;
import com.refinedmods.refinedstorage.container.InterfaceContainerMenu;
import com.refinedmods.refinedstorage.screen.widget.sidebutton.ExactModeSideButton;
import com.refinedmods.refinedstorage.screen.widget.sidebutton.RedstoneModeSideButton;
import net.minecraft.client.gui.GuiGraphics;
import net.minecraft.client.resources.language.I18n;
import net.minecraft.network.chat.Component;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.world.entity.player.Inventory;

public class InterfaceScreen extends BaseScreen<InterfaceContainerMenu> {
    private static final ResourceLocation TEXTURE = new ResourceLocation(RS.ID, "textures/gui/interface.png");

    public InterfaceScreen(InterfaceContainerMenu containerMenu, Inventory inventory, Component title) {
        super(containerMenu, 211, 217, inventory, title);
    }

    @Override
    public void onPostInit(int x, int y) {
        addSideButton(new RedstoneModeSideButton(this, NetworkNodeBlockEntity.REDSTONE_MODE));

        addSideButton(new ExactModeSideButton(this, InterfaceBlockEntity.COMPARE));
    }

    @Override
    public void tick(int x, int y) {
        // NO OP
    }

    @Override
    public void renderBackground(GuiGraphics graphics, int x, int y, int mouseX, int mouseY) {
        graphics.blit(TEXTURE, x, y, 0, 0, imageWidth, imageHeight);
    }

    @Override
    public void renderForeground(GuiGraphics graphics, int mouseX, int mouseY) {
        renderString(graphics, 7, 7, I18n.get("gui.refinedstorage.interface.import"));
        renderString(graphics, 7, 42, I18n.get("gui.refinedstorage.interface.export"));
        renderString(graphics, 7, 122, I18n.get("container.inventory"));
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/screen/InterfaceScreen.kt
package com.refinedmods.refinedstorage.screen

import com.mojang.blaze3d.matrix.MatrixStack
import com.refinedmods.refinedstorage.RS
import com.refinedmods.refinedstorage.container.InterfaceContainer
import com.refinedmods.refinedstorage.screen.widget.sidebutton.ExactModeSideButton
import com.refinedmods.refinedstorage.screen.widget.sidebutton.RedstoneModeSideButton
import com.refinedmods.refinedstorage.tile.InterfaceTile
import net.minecraft.client.resources.I18n
import net.minecraft.entity.player.PlayerInventory
import net.minecraft.util.text.Text

class InterfaceScreen(container: InterfaceContainer, inventory: PlayerInventory?, title: Text?) : BaseScreen<InterfaceContainer?>(container, 211, 217, inventory, title) {
    override fun onPostInit(x: Int, y: Int) {
        addSideButton(RedstoneModeSideButton(this, InterfaceTile.REDSTONE_MODE))
        addSideButton(ExactModeSideButton(this, InterfaceTile.COMPARE))
    }

    override fun tick(x: Int, y: Int) {}
    override fun renderBackground(matrixStack: MatrixStack?, x: Int, y: Int, mouseX: Int, mouseY: Int) {
        bindTexture(RS.ID, "gui/interface.png")
        blit(matrixStack, x, y, 0, 0, xSize, ySize)
    }

    override fun renderForeground(matrixStack: MatrixStack?, mouseX: Int, mouseY: Int) {
        renderString(matrixStack, 7, 7, I18n.format("gui.refinedstorage.interface.import"))
        renderString(matrixStack, 7, 42, I18n.format("gui.refinedstorage.interface.export"))
        renderString(matrixStack, 7, 122, I18n.format("container.inventory"))
    }
}